clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredLoginCallbackHandler.java

```diff
@@ -140,10 +151,12 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
         for (Callback callback : callbacks) {
             if (callback instanceof OAuthBearerTokenCallback)
                 try {
-                    handleCallback((OAuthBearerTokenCallback) callback);
+                    handleTokenCallback((OAuthBearerTokenCallback) callback);
                 } catch (KafkaException e) {
                     throw new IOException(e.getMessage(), e);
                 }
+            else if (callback instanceof SaslExtensionsCallback)
+                handleExtensionsCallback((SaslExtensionsCallback) callback);
```

Comment:
Probably a good idea to wrap in a try {} catch (KafkaException e) {} block as is done above for the OAuthBearerTokenCallback.

Reply:
And just swallow the exception? I guess it boils down to:
Do we want to stop authentication on invalid extension value or just not use extensions?